### PR TITLE
Add markdown lint make rule

### PR DESCRIPTION
**Pull request by mattnibs**
Add (light) markdown linting from
https://github.com/igorshubovych/markdownlint-cli. Adjust github
workflow so this is run when markdown files are editted.

### CODE REVIEW

File: .markdownlint.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+default: false
+
+# Tags
+spelling: true
+indentation: false
+links: true
+whitespace: true
+html: true
```
Comment (nwt) [inline .markdownlint.yaml:8]: Nit: Alphabetize!

File: .gitignore [hunk 2]
```diff
@@ -2,6 +2,7 @@
 /compiler/parser/deps/
 /dist/
 /zed-sample-data/
+/node_modules/
```
Comment (nwt) [inline .gitignore:5]:
```suggestion
/zed-sample-data/
/node_modules/
```
```suggestion
/node_modules/
/zed-sample-data/
```

File: .markdownlint.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+# Individual tweaks
+MD010:
+  code_blocks: false # Disallow hard tabs except in code blocks.
+MD028: false # Allow blank lines between blockquotes.
```
Comment (nwt) [inline .markdownlint.yaml:13]: I'm for dropping this one.

File: .github/workflows/markdown-lint.yml [hunk 4]
```diff
@@ -1,4 +1,4 @@
-name: Markdown link check
+name: Markdown lint
```
Comment (nwt) [inline .github/workflows/markdown-lint.yml:1]: s/link-check/lint/ on lines 12 and 32.

**Review by nwt — APPROVED** [review]

File: .markdownlint.yaml (hunk 3 above)
Comment (mattnibs) [reply .markdownlint.yaml:13]: You mean allowing this one?

Comment (nwt) [reply .markdownlint.yaml:13]: I mean that I think we should disallow blank lines between blockquotes (and fix the few that we have).

Comment (mattnibs) [reply .markdownlint.yaml:13]: Allowing this means you can't do:

> A block quote

> A consecutive block quote

There was a couple places where we are doing this and it seemed appropriate, but am happy to go with your recommendation.

**Pull request merged**